helper/ttl/validation.go

```diff
@@ -0,0 +1,83 @@
+package ttl
+
+import (
+	"errors"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/vault/logical"
+)
+
+type MountHandler struct {
+	// ConfigTTL is the TTL being set at a mount's config level.
+	// For example, if your plugin were the Azure secrets engine,
+	// and you had a path for an overall config like <mount>/config,
+	// this would be the TTL at that level.
+	// Optional, can be unset.
+	ConfigTTL time.Duration
+
+	// ConfigMaxTTL is the MaxTTL being set at a mount's config level.
+	// For example, if your plugin were the Azure secrets engine,
+	// and you had a path for an overall config like <mount>/config,
+	// this would be the MaxTTL at that level.
+	// Optional, can be unset.
+	ConfigMaxTTL time.Duration
+
+	// RoleTTL is the TTL being set at a role level, which is a lower
+	// and more specific level than the config level.
+	// Optional, can be unset.
+	RoleTTL time.Duration
+
+	// RoleMaxTTL is the MaxTTL being set at a role level, which is a lower
+	// and more specific level than the config level.
+	// Optional, can be unset.
+	RoleMaxTTL time.Duration
+}
+
+func (h *MountHandler) Validate(system logical.SystemView) error {
+
+	merr := &multierror.Error{}
+
+	// Verify the config-level TTL's alone.
+	if h.ConfigTTL < 0 {
+		merr = multierror.Append(merr, errors.New("config ttl < 0"))
+	}
+	if h.ConfigMaxTTL < 0 {
```

Comment:
Could we instead add this check to the type system for TypeDuration? Would remove 4 of the checks in this function. https://github.com/hashicorp/vault/blob/master/logical/framework/field_data.go#L229

Reply:
We shouldn't want negative durations anywhere in vault, and if we do we can make another type for them

Reply:
We should do some due diligence checking existing uses of typedurationsecond but I don't offhand remember any places where negative durations are allowed.